core/src/main/java/haveno/core/api/CoreApi.java

```diff
@@ -345,7 +345,11 @@ public int getNumConfirmationsForMostRecentTransaction(String addressString) {
     }
 
     public long getHeight() {
-        return  walletsService.getHeight();
+        return walletsService.getHeight();
+    }
+
+    public long getTargetHeight() {
+        return walletsService.getTargetHeight();
```

Comment:
CoreApi has `xmrConnectionService`, so we can return `Objects.requireNonNullElse(xmrConnectionService.getTargetHeight(), 0L);` directly here without adding a call to CoreWalletService or XmrWalletService.